pkg/solana/logpoller/log_poller.go

```diff
@@ -44,15 +68,94 @@ func New(lggr logger.SugaredLogger, orm ORM) *LogPoller {
 		Start: lp.start,
 	}.NewServiceEngine(lggr)
 	lp.lggr = lp.eng.SugaredLogger
+
 	return lp
 }
 
 func (lp *LogPoller) start(context.Context) error {
+	if lp.typeProvider == nil {
```

Comment:
Type provider is always nil

Reply:
Yeah, I replaced `lp.typeProvider` just last night with a `decoder` object attached to each filter, just forgot to remove it.